### PR TITLE
zamir/van 1295/add tests for recommender

### PR DESCRIPTION
**Pull request by zainab-amir**
### Description

Add tests for the recommendations functionality. Tests added for
- Progressive profiling page: This is where we decide whether to add recommendations or not
- Recommendations page tests
- 

#### Merge Checklist

* [ ] Is there adequate test coverage for your changes?

### CODE REVIEW

File: src/recommendations/RecommendationsPage.jsx [hunk 1]
```diff
@@ -47,7 +50,7 @@ const RecommendationsPage = (props) => {
     }
   };
 
-  if (!isLoading && recommendations.length < 3) {
+  if (!isLoading && recommendations.length < RECOMMENDATIONS_COUNT) {
```
Comment (syedsajjadkazmii) [inline src/recommendations/RecommendationsPage.jsx:53]: Don't we want to show 3 recommendations?

File: src/recommendations/tests/RecommendationsPage.test.jsx [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+const IntlRecommendationsPage = injectIntl(RecommendationsPage);
+const mockStore = configureStore();
+
+jest.mock('../data/service');
+
+describe('RecommendationsPageTests', () => {
+  let defaultProps = {};
+  let store = {};
+
+  const mockedResponse = [
```
Comment (syedsajjadkazmii) [inline src/recommendations/tests/RecommendationsPage.test.jsx:23]: can you put this mocked constant data into a separate file such as `tests/constants.js`. This approach is used by learner home and I think its a good approach to separate constants from tests.

File: src/recommendations/RecommendationsPage.jsx (hunk 1 above)
Comment (zainab-amir) [reply src/recommendations/RecommendationsPage.jsx:53]: I am not sure where this `3` came from so for safe side I have used `RECOMMENDATIONS_COUNT` and have also created a thread with UX for confirmation. They are finalizing a strategy where we will backfill the missing recommendations with general ones but for now lets keep it the condition as mentioned above.

**Review by syedsajjadkazmii — COMMENTED** [review]
I don't see any test for the loader state. could you please add that too?

**Review by syedsajjadkazmii — APPROVED** [review]
Thanks

**Pull request merged**